bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt

```diff
@@ -84,11 +87,20 @@ abstract class ImportIntoModelServer @Inject constructor(of: ObjectFactory) : De
         runBlocking {
             client.init()
             client.runWrite(branchRef) { rootNode ->
-                logger.info("Got root node: {}", rootNode)
-                logger.info("Importing...")
-                ModelImporter(rootNode, continueOnError.get()).importFilesAsRootChildren(files)
-                logger.info("Import finished")
+                rootNode.runBulkUpdate {
+                    logger.info("Got root node: {}", rootNode)
+                    logger.info("Importing...")
+                    ModelImporter(rootNode, continueOnError.get()).importFilesAsRootChildren(files)
+                    logger.info("Import finished")
+                }
             }
         }
     }
 }
+
+/**
+ * Memory optimization that doesn't record individual change operations, but only the result.
+ */
+private fun INode.runBulkUpdate(body: () -> Unit) {
+    ((this as PNodeAdapter).branch as OTBranch).runBulkUpdate(body = body)
```

Comment:
Is there a chance that this cast can fail?

Reply:
It's a private method that is only used once on the INode created by `IModelClientV2.runWrite`. If it succeeds once it will always succeed. Unit tests should detect any incompatible changes in the implementation of `IModelClientV2.runWrite`.